Sources/StreamChatUI/ChatMessageList/ChatTypingIndicatorView.swift

```diff
@@ -0,0 +1,150 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+// TODO: Decide which IndicatorView to Use:
+// A) AChatTypingIndicatorView
+// B) ChatTypingIndicatorView
+
+/// A `UIView` subclass with 3 dots which can be animated with fading out effect.
+public typealias AChatTypingIndicatorView = _ChatTypingIndicatorView<NoExtraData>
+
+/// A `UIView` subclass with 3 dots which can be animated with fading out effect.
+public typealias ChatTypingIndicatorView = _ChatTypingIndicatorView<NoExtraData>
+
+/// A `UIView` subclass with 3 dots which can be animated with fading out effect.
+open class _ChatTypingIndicatorView<ExtraData: ExtraDataTypes>: _View, UIConfigProvider {
+    open var dotSize: CGSize = .init(width: 5, height: 5)
+
+    private(set) lazy var innerView: UIView = UIView().withoutAutoresizingMaskConstraints
```

Comment:
why do we need this?

Reply:
I could achieve this other ways mentioned:
 1.  Adding some borders to the layer in the view. 
 2. Using frame based layout for the layer inside the view.  

However IMHO is the most easier way to add `insets` to the layer and add the replicator layer inside with no fear it will touch the leading and trailing. Because on layer, you cannot add the insets.

Reply:
Okay I deleted the innerView because it was quite complex (and ugly) to add the margins to it and make it work properly. Now the view has no margins and they need to be setup manually from outside (which IMHO makes more sense)